kafka/consumer/main.go

```diff
@@ -123,15 +138,16 @@ func main() {
 	}
 
 	for {
-		// Will wait up to 1 minute for 100 events. If the context is cancelled (or expires)
+		// Will wait up to 10 seconds for 100 events. If the context is cancelled (or expires)
 		// you'll get any events that have been collected up to that point.
-		receiveCtx, cancel := context.WithTimeout(context.TODO(), time.Second*10)
-		log.Println("Start to receive events ")
+		receiveCtx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		log.Println("Start to receive events.")
 		events, err := partitionClient.ReceiveEvents(receiveCtx, 100, nil)
 		cancel()
 
 		if err != nil && !errors.Is(err, context.DeadlineExceeded) {
-			panic(err)
+			log.Printf("Receiving events failed due to context deadline exceeded: %s", err.Error())
```

Comment:
The log message says that the failure is due to the context deadline being exceeded, but the if-condition  filters out this specific error.